statistics/handle/handle_hist.go

```diff
@@ -0,0 +1,308 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handle
+
+import (
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/metrics"
+	"github.com/pingcap/tidb/parser/model"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessionctx/stmtctx"
+	"github.com/pingcap/tidb/statistics"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/logutil"
+	"github.com/pingcap/tidb/util/sqlexec"
+	"github.com/pingcap/tidb/util/timeutil"
+	"go.uber.org/zap"
+)
+
+// TODO load idx histograms by need
+
+type StatsLoad struct {
+	sync.Mutex
+	SubCtxs          []sessionctx.Context
+	NeededColumnsCh  chan *NeededColumnTask
+	TimeoutColumnsCh chan *NeededColumnTask
+	workingColMap    map[model.TableColumnID]struct{}
+}
+
+// NeededColumnTask represents one needed column with expire time.
+type NeededColumnTask struct {
+	TableColumnID model.TableColumnID
+	ToTimeout     time.Time
+	Wg            *sync.WaitGroup
```

Comment:
WaitGroup needs to be passed as a reference, otherwise, it's a copy and cannot work properly. 
https://stackoverflow.com/questions/27893304/golang-sync-waitgroup-never-completes